pkg/fanal/walker/walk.go

```diff
@@ -6,85 +6,47 @@ import (
 	"strings"
 
 	"github.com/bmatcuk/doublestar/v4"
+	"github.com/samber/lo"
 
 	"github.com/aquasecurity/trivy/pkg/fanal/analyzer"
-	"github.com/aquasecurity/trivy/pkg/fanal/utils"
 	"github.com/aquasecurity/trivy/pkg/log"
 )
 
-var (
-	// These variables are exported so that a tool importing Trivy as a library can override these values.
-	AppDirs    = []string{".git"}
-	SystemDirs = []string{
-		"proc",
-		"sys",
-		"dev",
-	}
-)
-
-const defaultSizeThreshold = int64(200) << 20 // 200MB
+const defaultSizeThreshold = int64(100) << 20 // 200MB
 
-type WalkFunc func(filePath string, info os.FileInfo, opener analyzer.Opener) error
-
-type walker struct {
-	skipFiles []string
-	skipDirs  []string
+var defaultSkipDirs = []string{
+	"**/.git",
```

Comment:
I have a question. Why you just ignore the .git folder?

Reply:
The git dir shouldn't have files Trivy wants to find, and it is time-consuming to walk the dir.

Reply:
> The git dir shouldn't have files Trivy wants to find, and it is time-consuming to walk the dir.

thank u for answering. But I want to say something. just like what github official does.

https://github.com/actions/checkout/blob/44c2b7a8a4ea60a981eaca3cf939b5f4305c123b/src/git-auth-helper.ts#L286

They allows save PAT tokens or request OAuth tokens in this actions. And backup this  So some careless people do like 

```dockerfile
FROM node:16
RUN mkdir /app
COPY . /app
RUN yarn install
EXPOSE 80
CMD ["node", "/app/app.js"]
```
And Python/PHP/Javascript programmers always do like this. : ) 

So at least whitelist the `.git/config` `.git/modules/[any]/config` and blacklist something like `.git/objects`

Reply:
```
FROM node:16
RUN mkdir /app
COPY . /app
RUN yarn install
EXPOSE 80
CMD ["node", "/app/app.js"]
```

What's wrong in this Dockerfile? PAT will be deleted in the post-job step.

Reply:
> ```
> FROM node:16
> RUN mkdir /app
> COPY . /app
> RUN yarn install
> EXPOSE 80
> CMD ["node", "/app/app.js"]
> ```
> 
> What's wrong in this Dockerfile? PAT will be deleted in the post-job step.

Yes post job. But docker image building is before it :) 
It's 

```mermaid
graph TD;
START ---> checkout ---> docker-image-building ---> docker-image-uploading ---> post-docker-login ----> post-checkout  ---> deleted-PAT ---> github.token-expired
```

Reply:
Hmm. It's interesting. Did you ensure PAT can be exposed to a container image?

Reply:
> Hmm. It's interesting. Did you ensure PAT can be exposed to a container image?

pretty sure. Play with cross repos ref submodules  they must do like that. I can create an environment. : )

Reply:
I've created an issue. Thanks!
https://github.com/aquasecurity/trivy/issues/6699

Reply:
> I've created an issue. Thanks! #6699

nice work :)